packages/loot-core/src/server/budget/goaltemplates.ts

```diff
@@ -97,22 +91,27 @@ async function processTemplate(month, force, category_templates) {
             : lowestPriority;
       }
     }
-  }
-
-  await setZero({ month });
-
-  //setZero() sets budgeted Income to 0. Reset income categories before continuing.
-  if (isReflectBudget()) {
-    for (let l = 0; l < originalCategoryBalance.length; l++) {
-      if (originalCategoryBalance[l].isIncome) {
-        await setBudget({
-          category: originalCategoryBalance[l].cat.id,
-          month,
-          amount: originalCategoryBalance[l].amount,
-        });
-      }
+    if (budgeted) {
```

Comment:
This is creating a list of of all categories in the budget along with the currently filled in amount and whether or not it has a template.